Bottom: 6383b51df66b84ec6542ad5840b836c9b1c5729e
Top:    dd20c02a9621167bf54aa37a2ba3e1a778fe9df7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 19:16:17 +0000

Fixup model orientation harder

Be consistent with earlier work without the REVERSED_PCB_CONTOURS flag unset.


---

diff --git a/src/hid/step/step.c b/src/hid/step/step.c
index 81c11f7..76ab121 100644
--- a/src/hid/step/step.c
+++ b/src/hid/step/step.c
@@ -384,26 +384,35 @@ step_do_export (HID_Attr_Val * options)
         instance = g_new0 (struct assembly_model_instance, 1);
         instance->name = NAMEONPCB_NAME (element);
 #ifdef REVERSED_PCB_CONTOURS
-        instance->ox = COORD_TO_STEP_X (PCB, element->MarkX) +                    ( ox * cos_rot + oy * sin_rot),
-        instance->oy = COORD_TO_STEP_Y (PCB, element->MarkY) - on_solder_negate * (-ox * sin_rot + oy * cos_rot),
-        instance->oz = (on_solder ? -COORD_TO_STEP_Z (PCB, HACK_BOARD_THICKNESS) : 0.0) - on_solder_negate * oz;
-        instance->ax =                    ( ax * cos_rot + ay * sin_rot),
-        instance->ay = -on_solder_negate * (-ax * sin_rot + ay * cos_rot),
-        instance->az = on_solder_negate * az;
-        instance->rx =                    ( rx * cos_rot + ry * sin_rot), /* XXX: SHOULD THIS FACTOR IN on_solder_negate? */
-        instance->ry = -on_solder_negate * (-rx * sin_rot + ry * cos_rot), /* XXX: SHOULD THIS FACTOR IN ol_solder_negaet? */
-        instance->rz = on_solder_negate * rz;
+        instance->ox =                     ( ox * cos_rot + oy * sin_rot);
+        instance->oy = -on_solder_negate * (-ox * sin_rot + oy * cos_rot);
+        instance->oz = -on_solder_negate * oz; /* <--- ????: -ve on on_solder_negative seems inconsistent w.r.t. others! */
+        instance->ax =                     ( ax * cos_rot + ay * sin_rot);
+        instance->ay = -on_solder_negate * (-ax * sin_rot + ay * cos_rot);
+        instance->az = -on_solder_negate * az;
+        instance->rx =                     ( rx * cos_rot + ry * sin_rot);
+        instance->ry = -on_solder_negate * (-rx * sin_rot + ry * cos_rot);
+        instance->rz = -on_solder_negate * rz;
 #else
-        instance->ox = COORD_TO_STEP_X (PCB, element->MarkX) +                    ( ox * cos_rot + oy * sin_rot),
-        instance->oy = COORD_TO_STEP_Y (PCB, element->MarkY) + on_solder_negate * (-ox * sin_rot + oy * cos_rot),
-        instance->oz = on_solder_negate * -COORD_TO_STEP_Z (PCB, HACK_BOARD_THICKNESS) / 2.0 + on_solder_negate * oz;
-        instance->ax =                    ( ax * cos_rot + ay * sin_rot),
-        instance->ay = on_solder_negate * (-ax * sin_rot + ay * cos_rot),
-        instance->az = on_solder_negate * az;
-        instance->rx =                    ( rx * cos_rot + ry * sin_rot), /* XXX: SHOULD THIS FACTOR IN on_solder_negate? */
-        instance->ry = on_solder_negate * (-rx * sin_rot + ry * cos_rot), /* XXX: SHOULD THIS FACTOR IN ol_solder_negaet? */
-        instance->rz = on_solder_negate * rz;
+        instance->ox =                     ( ox * cos_rot + oy * sin_rot);
+        instance->oy =  on_solder_negate * (-ox * sin_rot + oy * cos_rot);
+        instance->oz =  on_solder_negate * oz;
+        instance->ax =                     ( ax * cos_rot + ay * sin_rot);
+        instance->ay =  on_solder_negate * (-ax * sin_rot + ay * cos_rot);
+        instance->az =  on_solder_negate * az;
+        instance->rx =                     ( rx * cos_rot + ry * sin_rot);
+        instance->ry =  on_solder_negate * (-rx * sin_rot + ry * cos_rot);
+        instance->rz =  on_solder_negate * rz;
 #endif
+
+        instance->ox += COORD_TO_STEP_X (PCB, element->MarkX);
+        instance->oy += COORD_TO_STEP_Y (PCB, element->MarkY);
+#ifdef REVERSED_PCB_CONTOURS
+        instance->oz += COORD_TO_STEP_Z (PCB, on_solder ? -HACK_BOARD_THICKNESS : 0);
+#else
+        instance->oz += COORD_TO_STEP_Z (PCB, on_solder_negate * -HACK_BOARD_THICKNESS / 2);
+#endif
+
         model->instances = g_list_append (model->instances, instance);
 
       }
